### ISSUE TITLE
X86 build fails on Flatpak

### ISSUE THREAD

**Issue by coslyk** [annulen/webkit#795]
You can see the build logs here: https://flathub.org/builds/#/builders/2/builds/7838

Error is
```
../Source/WebCore/bindings/js/JSDOMConstructor.h:195:35: error: invalid conversion from ‘JSC::EncodedJSValue (*)(JSC::ExecState*)’ {aka ‘long long int (*)(JSC::ExecState*)’} to ‘JSC::NativeFunction’ {aka ‘long long int (__attribute__((fastcall)) *)(JSC::ExecState*)’} [-fpermissive]
     constructData.native.function = construct;
     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~
```

**Comment by annulen** [comment]
It seems to me that you fixed the issue with
```diff
--- a/Source/WTF/wtf/Platform.h
+++ b/Source/WTF/wtf/Platform.h
@@ -158,6 +158,7 @@
 /* CPU(X86) - i386 / x86 32-bit */
 #if   defined(__i386__) \
     || defined(i386)     \
+    || defined(__i386)   \
     || defined(_M_IX86)  \
     || defined(_X86_)    \
     || defined(__THW_INTEL)
```
However, in WebKit upstream there is no such change. I guess that other WebKit ports, in particular WebKitGTK, should be affected by the same issue.

**Comment by annulen** [comment]
Do you have any idea why __i386__ is not defined in your environment?

**Comment by coslyk** [comment]
Adding __i386 still does not fix the issue. Maybe there is some conflicts with other header files that make macro `JSC_HOST_CALL` not be set correctly.

**Comment by annulen** [comment]
Seems like this issue is related to gcc 8, not flatpack

**Comment by mitya57** [comment]
I can confirm this issue on Debian and Ubuntu. It happens only on X86 when this define is active:

https://github.com/annulen/webkit/blob/4ce8ebc4094512b9916bfa5984065e95ac97c9d8/Source/WTF/wtf/Platform.h#L838-L839

Adding `-fpermissive` helps, but I would rather avoid that.

**Comment by mitya57** [comment]
In upstream this was changed in a925f8a98ffbe762059ec91c5864da864946faff.

**Comment by annulen** [comment]
Thanks for heads up!

**Comment by mitya57** [comment]
The just filed PR #803 should fix this.

**Comment by coslyk** [comment]
> The just filed PR #803 should fix this.

Thanks for your patch! It works!
Build log: https://flathub.org/builds/#/builders/32/builds/1556
Now the new error occurs due to the lack of memory, but it is the problem with the build environment, not with codes.

**Comment by annulen** [comment]
I've merged #803, is this issue resolved now?

**Comment by coslyk** [comment]
@annulen Yes. Thanks!
